### PR TITLE
Destroy FilterState when object destroyed #3973

### PR DESCRIPTION
**Pull request by YahorPadlucki**
Examples
**pixijs 4.7.1:**
https://yahorpadlucki.github.io/pixiExample/default.html
image

**this PR:**
https://yahorpadlucki.github.io/pixiExample/custom.html
image

Seems to fix this issue - but not sure if I am doing it right?

### CODE REVIEW

**Comment by YahorPadlucki** [comment]
Hi, @bigtimebuddy @themoonrat @ivanpopelyshev 
can somebody look at it? Some how it is the only PR without comments ( Did I prepare it wrong?

**Comment by englercj** [comment]
I don't think this is the right way to fix this issue. Having every container hold a ref to the last renderer to operate on it seems like a code smell.

Instead I would have objects notify that they are being destroyed and have the filter manager listening for those events from all the targets it works on. It can then cleanup any state needed.

**Comment by YahorPadlucki** [comment]
Thanks a lot for explanation, Chad ! 
Will change it.

**Comment by YahorPadlucki** [comment]
Updated with 'destroyed' event listener

**Comment by englercj** [comment]
I think this solution a lot more. Thanks for iterating on that feedback!

File: src/core/renderers/webgl/managers/FilterManager.js [hunk 1]
```diff
@@ -53,6 +65,7 @@ export default class FilterManager extends WebGLManager
         this.filterData = null;
 
         this.managedFilters = [];
+        this.managedDisplayObjects = [];
```
Comment (bigtimebuddy) [inline src/core/renderers/webgl/managers/FilterManager.js]: Would we maybe use something a little less verbose like `managedTargets`?

File: src/core/renderers/webgl/managers/FilterManager.js [hunk 2]
```diff
@@ -537,6 +556,41 @@ export default class FilterManager extends WebGLManager
...
+        });
+        this.managedDisplayObjects = [];
+    }
+
+    /**
+     * Destroys FilterState of specified target
+     *
+     * @param {PIXI.DisplayObject} target, which FilterState should be destroyed
+     */
+    destroyFilterStateByTarget(target)
```
Comment (bigtimebuddy) [inline src/core/renderers/webgl/managers/FilterManager.js]: Why create a separate function? Wouldn't it just be easier to inline in the destroy handler?

File: src/core/renderers/webgl/managers/FilterManager.js [hunk 3]
```diff
@@ -666,4 +720,20 @@ export default class FilterManager extends WebGLManager
             this.pool[screenKey] = [];
         }
     }
+
+    /**
+     * Called when managed DisplayObject is destroyed
+     *
+     * @param {PIXI.DisplayObject} destroyedDisplayObject, object which was destroyed
+     */
+    onManagedDisplayObjectDestroyed(destroyedDisplayObject)
```
Comment (bigtimebuddy) [inline src/core/renderers/webgl/managers/FilterManager.js]: Less verbose name suggestion: `onTargetDestroyed`

**Comment by YahorPadlucki** [comment]
@bigtimebuddy , Thanks a lot for review. 
Made changes according to review, added unit-test for destroyed event

**Comment by bigtimebuddy** [comment]
Closing in favor of #4897

**Pull request closed without merging**